src/lib/study-webidl.js

```diff
@@ -0,0 +1,199 @@
+const { expandCrawlResult } = require("reffy");
+const fetch = require("node-fetch");
+const WebIDL2 = require("webidl2");
+
+const report = {};
+
+// DRY with study-backref
+function recordAnomaly(spec, anomalyType, error) {
+  if (!report[spec.url]) {
+    report[spec.url] = {
+      title: spec.title,
+      crawled: spec.crawled,
+      shortname: spec.shortname,
+      repo: spec.nightly.repository,
+      // specific to this report
+      invalidWebIdl: [],
+      idlNameReused: [],
+      noMainIdlDefinition: [],
+      noIdlExposure: [],
+      incompatiblePartialIdlExposure: [],
+      duplicateIdlDefinition: [],
+      unexpectedEventHandler: [],
+      singleEnumValue: [],
+      wrongCaseEnumValue: []
```

Comment:
Looking into this, from a structuring perspective, I would have gone with a single array `webidl` that contains a list of anomaly objects à la:

```json
{
  "type": "...",
  "msg": "..."
}
```

Use of a single property would make it easier to merge anomaly reports (webidl, backrefs, etc.) together without having to introduce another level to group anomalies according to study type. The code already treats all anomalies similarly, it seems good to reflect that in the resulting structure.

Use of an object would make it easier to extend the reports over time with info such as line number, link to documentation, perhaps a sugested fix, etc.

That said, I see that `study-backrefs.js` follows the same approach as here, so definitely something that could be done later on.

Reply:
I think it's indeed worth revisiting the structure used for the reports, but I further agree that we should do that across the codebase, so let's maybe merge this as is and take a broader look at the issue reporter and integration with webref tests?

Reply:
Not sure thumbs up reactions get reported as notifications, so in case you didn't see it: 👍.